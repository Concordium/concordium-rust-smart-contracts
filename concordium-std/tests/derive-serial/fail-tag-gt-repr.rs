//! Ensure `derive(Serial)` fails when a 'tag' attribute is greater than what
//! 'repr(u8)' can represent.
use concordium_std::*;

#[derive(Serial)]
#[concordium(repr(u8))]
enum MyTaggedEnum {
    #[concordium(tag = 256)]
    One,
    Two,
}

fn main() {}
