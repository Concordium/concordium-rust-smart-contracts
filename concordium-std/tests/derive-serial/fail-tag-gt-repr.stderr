error: 'tag' attribute value of 256 cannot be represented by the type u8 set in 'repr(u8)'
 --> tests/derive-serial/fail-tag-gt-repr.rs:8:18
  |
8 |     #[concordium(tag = 256)]
  |                  ^^^
